setup.py

```diff
@@ -79,5 +79,10 @@ def _list_files(path, match):
         "pytest",
         "pyyaml",
         "requests",
-    ]
+    ],
+    extras_require={
+        'dev': [
+            'pycodestyle==2.8.0'
```

Comment:
I believe the convention is to not specify package versions in setup.py but to do it in requirements.txt instead.

Reply:
I guess that limiting this version to a single one is too tight, I removed the version.

Having said that it's not exactly that specifying package version in setup.py is an anti-pattern. The setup.py should specify a minimal set of requirements to run the application, so should it be running with any version of the dependency there should be no version, but it's perfectly ok to specify a version or a range of versions for a package when e.g. API changed. The dependencies/requirements definition in setup.py doesn't also specify a source of the packages.

Reply:
I mean, all the other packages don't have any version specified.  So if we want to maintain versions both in requirements.txt and setup.py then we should add version information for all the other packages too in setup.py and always ensure they're kept in sync with requirements.txt.  That seems like some overhead to me for not much added value.

Reply:
setup.py and requirements.txt serve for a slightly different purposes. It's quite well described [here](https://packaging.python.org/en/latest/discussions/install-requires-vs-requirements/). I think it's better to keep setup.py without version sepcified or with minimum/maximum version specified only when other can break the application.